txmgr: Disable default batcher tx send timeout

sebastianst opened the pull request:
With Holocene, batcher transaction ordering has to be strictly preserved, so trying to send a transaction candidate should just never timeout. Note that the txmgr will still bump fees to get a transaction for the same nonce submitted.

geoknee commented on op-service/txmgr/cli.go:
```diff
@@ -79,7 +79,6 @@ var (
 		MinBaseFeeGwei:            1.0,
 		ResubmissionTimeout:       48 * time.Second,
 		NetworkTimeout:            10 * time.Second,
-		TxSendTimeout:             10 * time.Minute,
```
How about we explicitly set it to `nil` and add a comment explaining that it is due to Holocene rules?

sebastianst committed:
txmgr: Disable default batcher tx send timeout

With Holocene, batcher transaction ordering has to be strictly preserved, so
trying to send a transaction candidate should just never timeout.
Note that the txmgr will still bump fees to get a transaction for the
same nonce submitted.

tynes commented on op-service/txmgr/cli.go:
```diff
@@ -79,7 +79,7 @@ var (
 		ResubmissionTimeout:       48 * time.Second,
 		NetworkTimeout:            10 * time.Second,
-		TxSendTimeout:             10 * time.Minute,
+		TxSendTimeout:             0, // Try sending txs indefinitely, to preserve tx ordering for Holocene
```
Should this be an override specific to the batcher and not for the generic service?

sebastianst replied:
@tynes It is specific for the batcher, it's the `DefaultBatcherFlagValues` struct that I'm changing here. It's slightly confusing that defaults for different specific services live in the general `op-service/txmgr` module.
